specification/applicationinsights/ApplicationInsights.LiveMetrics/models.tsp

```diff
@@ -0,0 +1,567 @@
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model DocumentStreamInfo {
+  @doc("Identifier of the document stream initiated by a UX.")
+  Id: string;
+
+  @doc("Gets or sets an OR-connected collection of filter groups.")
+  DocumentFilterGroups: DocumentFilterConjunctionGroupInfo[];
+}
+
+/**
+ * A collection of filters for a specific telemetry type.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model DocumentFilterConjunctionGroupInfo {
+  @doc("Telemetry type.")
+  TelemetryType:
+    | "Request"
+    | "Dependency"
+    | "Exception"
+    | "Event"
+    | "Metric"
+    | "PerformanceCounter"
+    | "Trace";
+
+  @doc("An array of filter groups.")
+  Filters: FilterConjunctionGroupInfo;
+}
+
+/**
+ * A metric configuration set by UX to scope the metrics it's interested in.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model DerivedMetricInfo {
+  @doc("metric configuration identifier.")
+  Id: string;
+
+  @doc("Telemetry type.")
+  TelemetryType: string;
+
+  @doc("A collection of filters to scope metrics that UX needs.")
+  FilterGroups: FilterConjunctionGroupInfo[];
+
+  @doc("Telemetry's metric dimension whose value is to be aggregated. Example values: Duration, Count(),...")
+  Projection: string;
+
+  @doc("Aggregation type. This is the aggregation done from everything within a single server.")
+  Aggregation: "Avg" | "Sum" | "Min" | "Max";
+
+  @doc("Aggregation type. This Aggregation is done across the values for all the servers taken together.")
+  BackEndAggregation: "Avg" | "Sum" | "Min" | "Max";
+}
+
+/**
+ * An AND-connected group of FilterInfo objects.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model FilterConjunctionGroupInfo {
+  @doc("An array of filters.")
+  Filters: FilterInfo[];
+}
+
+/**
+ * A filter set on UX
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+model FilterInfo {
+  @doc("dimension name of the filter")
+  FieldName: string;
+
+  @doc("Operator of the filter")
+  Predicate:
+    | "Equal"
+    | "NotEqual"
+    | "LessThan"
+    | "GreaterThan"
+    | "LessThanOrEqual"
+    | "GreaterThanOrEqual"
+    | "Contains"
+    | "DoesNotContain";
+
+  @doc("Comparand of the filter")
+  Comparand: string;
+}
+
+/**
+ * Optional http response body, whose existence carries additional error descriptions.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Need to follow the casing style of the existing SDK contract."
+@error
+model ServiceError {
+  @doc("Error code.")
+  @statusCode
+  _: 400 | 401 | 403 | 404 | 500 | 503;
```

Comment:
We don't generally document error codes unless there is good reason. The service may return any error code, or a transport along the way.